Enable Tailscale's builtin inbound HTTPS proxy (#137)
hassio-addons/addon-tailscale: tailscale/DOCS.md

<https://login.tailscale.com/start>

<details>

> <summary>

Reviewer: Let's not do those weird collapsable, this is not a 10 pages doc.
The fact you have to write `(click to open)` is already a good indication it is bad ux.

<i>Additional prerequisites if you want to access your Home Assistant device
with a Tailscale provided certificate, ie. through the Tailscale Proxy feature
<b>(click to open)</b></i>
</summary>

> ### Home Assistant configuration for Tailscale Proxy feature

Reviewer: Let's move this section after the installation procedure

You must configure Home Assistant to **not** use SSL certificates, to be
accessible through plain http connection. The Tailscale https Proxy will access
Home Assistant through `localhost` and will not accept a real certificate,
connection will be closed with `proxy error: x509: cannot validate certificate

> for 127.0.0.1 because it doesn't contain any IP SANs`

Reviewer: This is the default, I think just mention that Home Assistant itself should not use certificates and link to the official documentation is enough.

If you still want to use another https connection to access Home Assistant
through another network, please use the **NGINX Home Assistant SSL proxy**

> add-on.

Reviewer: Just stating to use a reverse proxy should be enough. I generally like to withhold recommendations of other add-ons.

So these lines have to be removed from your `/config/configuration.yaml`:

```yaml
http:
#  ssl_certificate: /ssl/fullchain.pem
#  ssl_key: /ssl/privkey.pem

> ```

Reviewer: These are not there by default, the example is even unneeded, as `http` is a default loaded component.

This section can be removed.

> Since Home Assistant by default blocks requests from proxies/reverse proxies,
> you need to tell your instance to allow requests from the Tailscale add-on.
> In order to do so, add the following lines to your `/config/configuration.yaml`
> without changing anything:

Reviewer: ```suggestion
Since Home Assistant, by default, blocks requests from proxies/reverse proxies,
you need to tell your instance to allow requests from the Tailscale add-on.
In order to do so, add the following lines to your `/config/configuration.yaml`
without changing anything:
```

```yaml
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 127.0.0.1
```

> **Note**: _There is no need to adapt anything in these lines since the addon

Reviewer: This is already mentioned in the section above. Let's remove it.

runs on your host network._

### Tailscale configuration for Tailscale Proxy feature

[DNS page][tailscale_dns]: Choose a **Tailnet name** and click **Enable HTTPS**
under HTTPS Certificates (see [Enabling HTTPS][tailscale_info_https] for more

> information)

Reviewer: This should go after the installation, it is not a pre-requisite.

Received files are stored in the `/share/taildrop` directory.

### Option: `proxy`

When not set, this option is enabled by default.

> Tailscale can provide a TLS certificate for your Home Assistant device within

Reviewer: ```suggestion
Tailscale can provide a TLS certificate for your Home Assistant instance within
```

your tailnet domain.

> This can prevent browsers to warn that HTTP URLs to your Home Assistant device
> look unencrypted (browsers are not aware of that connections between Tailscale
> nodes are secured with end-to-end encryption).

Reviewer: ```suggestion
This can prevent browsers from warning that HTTP URLs to your Home Assistant device
look unencrypted (browsers are not aware of the connections between Tailscale
nodes are secured with end-to-end encryption).
```

More information: [Enabling HTTPS][tailscale_info_https]

> 1. Configure Home Assistant to be accessible through HTTP connection (this is

Reviewer: ```suggestion
1. Configure Home Assistant to be accessible through an HTTP connection (this is
```

   the default). See [HTTP integration documentation][http_integration] for more
   information. If you still want to use another HTTPS connection to access Home
   Assistant, please use a reverse proxy add-on.

> 1. Home Assistant, by default, blocks requests from reverse proxies, like the
>    Tailscale Proxy. In order to enable it, add the following lines to your
>    `configuration.yaml`, without changing anything:

Reviewer: ```suggestion
1. Home Assistant, by default, blocks requests from reverse proxies, like the
   Tailscale Proxy. To enable it, add the following lines to your
   `configuration.yaml`, without changing anything:
```

   ```yaml
   http:
     use_x_forwarded_for: true
     trusted_proxies:
       - 127.0.0.1
   ```

1. Navigate to the [DNS page][tailscale_dns] of the admin console:

   - Choose a Tailnet name.

   - Enable MagicDNS if not already enabled.

   - Under HTTPS Certificates section, click Enable HTTPS.

1. Restart the add-on.

> **Note:** _You should not use any port number in the url that you used

Reviewer: ```suggestion
**Note:** _You should not use any port number in the URL that you used
```
